e2e/pages/adf/cardViewComponentPage.ts

```diff
@@ -193,4 +195,40 @@ export class CardViewComponentPage {
         });
     }
 
+    async clearDateField() {
+        const clearDateButton = element(by.css('mat-icon[data-automation-id="datepicker-date-clear-date"]'));
+        const dateValue = element(by.css('span[data-automation-id="card-date-value-date"]'));
+        await BrowserActions.click(clearDateButton);
+        return dateValue.getText();
+    }
+
+    async clearDateTimeField() {
+        const clearDateButton = element(by.css('mat-icon[data-automation-id="datepicker-date-clear-datetime"]'));
+        const dateValue = element(by.css('span[data-automation-id="card-datetime-value-datetime"]'));
+        await BrowserActions.click(clearDateButton);
+        return dateValue.getText();
+    }
+
+    enableClearDate() {
+        BrowserVisibility.waitUntilElementIsVisible(this.clearDateSwitch);
+
+        this.clearDateSwitch.getAttribute('class').then((check) => {
+            if (check.indexOf('mat-checked') === -1) {
+                this.clearDateSwitch.click();
+                expect(this.clearDateSwitch.getAttribute('class')).toContain('mat-checked');
```

Comment:
expects belong in the tests.
having expects in page objects is close to bad practice and should be avoided unless really REALLY needed.
is this expect really needed here? what are you trying to achieve by using it? can it be replaced with a waitUntil ...  ?